src/to_substrait.cpp

```diff
@@ -1391,6 +1430,90 @@ substrait::Rel *DuckDBToSubstrait::TransformIntersect(LogicalOperator &dop) {
 	return rel;
 }
 
+substrait::Rel *DuckDBToSubstrait::TransformCreateTable(LogicalOperator &dop) {
+	auto rel = new substrait::Rel();
+	auto &create_table = dop.Cast<LogicalCreateTable>();
+	auto &create_info = create_table.info.get()->Base();
+
+	auto schema = new substrait::NamedStruct();
+	auto type_info = new substrait::Type_Struct();
+	type_info->set_nullability(substrait::Type_Nullability_NULLABILITY_REQUIRED);
+	for (auto &name : create_info.columns.GetColumnNames()) {
+		schema->add_names(name);
+	}
+	for (auto &col_type : create_info.columns.GetColumnTypes()) {
+		auto s_type = DuckToSubstraitType(col_type, nullptr, false);
+		*type_info->add_types() = s_type;
+	}
+	schema->set_allocated_struct_(type_info);
+
+	if (create_table.children.size() == 0) {
+		// This is create table with schema
+		auto ddl = rel->mutable_ddl();
+		ddl->set_op(substrait::DdlRel::DdlOp::DdlRel_DdlOp_DDL_OP_CREATE);
+		ddl->set_object(substrait::DdlRel::DDL_OBJECT_TABLE);
+		ddl->set_allocated_table_schema(schema);
+
+		auto named_object = ddl->mutable_named_object();
+		named_object->add_names(create_info.schema);
+		named_object->add_names(create_info.table);
+		return rel;
+	}
+
+	// This is create table as select
+	substrait::Rel *input = nullptr;
+	switch (create_table.children[0]->type)
+	{
+	case LogicalOperatorType::LOGICAL_PROJECTION:
+		input = TransformProjection(*create_table.children[0]);
+		break;
+	default:
+		throw NotImplementedException("Create table with more than one child");
+	}
+	auto write = rel->mutable_write();
+	write->set_allocated_table_schema(schema);
+	write->set_allocated_input(input);
+	write->set_op(substrait::WriteRel::WriteOp::WriteRel_WriteOp_WRITE_OP_CTAS);
+	auto named_table = write->mutable_named_table();
+	named_table->add_names(create_info.schema);
+	named_table->add_names(create_info.table);
+
+	return rel;
+}
+
+substrait::Rel *DuckDBToSubstrait::TransformDropTable(LogicalOperator &dop) {
+	auto rel = new substrait::Rel();
+	auto &drop_table = dop.Cast<LogicalSimple>();
+	auto &drop_info = drop_table.info.get()->Cast<DropInfo>();
+
+	auto schema = new substrait::NamedStruct();
+	auto type_info = new substrait::Type_Struct();
+	type_info->set_nullability(substrait::Type_Nullability_NULLABILITY_REQUIRED);
+	schema->set_allocated_struct_(type_info);
+
+	auto ddl = rel->mutable_ddl();
+	switch (drop_info.type) {
+	case CatalogType::TABLE_ENTRY:
+		ddl->set_object(substrait::DdlRel::DDL_OBJECT_TABLE);
+		break;
+	case CatalogType::VIEW_ENTRY:
+		ddl->set_object(substrait::DdlRel::DDL_OBJECT_VIEW);
+	default:
+		throw NotImplementedException("Drop type not implemented");
```

Comment:
report type seen, don't assume that only drop lands here. will send people on wild goose chase/confusion in future.